spec/API_specification/array_api/fourier_transform_functions.py

```diff
@@ -0,0 +1,424 @@
+
+    Parameters
+    ----------
+    a: array
+        input array. Should have a floating-point data type.
+    s: Union[Sequence[int], Tuple[int, ...]]
+        size of each transformed axis of the **output**. If given, the last axis will be either zero-padded or trimmed to ``s[-1]//2+1``, whereas all other axes ``i`` are either zero-padded or trimmed to the length ``s[i]``, before computing the inverse of ``rfftn``. Otherwise, the last axis is either zero-padded or trimmed to ``2 * (m - 1)``, where `m` is the length of the input along the axis, and all other axes use the input shape. The last element ``s[-1]`` is for computing ``irfft(a[axes[-1]], n=s[-1])`` whereas other elements for ``ifft(a[axes[i]], n=s[i])``. Default: ``None``.
+    axes: Union[Sequence[int], Tuple[int, ...]]
+        axes over which to compute the inverse Fourier transform. If it is not specified, the last ``len(s)`` axes are used or all axes if ``s`` is also not specified. Default: ``None``.
+    norm: str
+        specify the normalization mode. Should be one of the following modes:
+
+        - ``'backward'``: Normalize by ``1/n``.
+        - ``'ortho'``: Normalize by ``1/sqrt(n)``
+        - ``'forward'``: No normalization.
+
+        Default: ``'backward'``.
+
+    Returns
+    -------
+    out: array
+        a real-valued array transformed along the axes indicated by the ``axes`` keyword. The length along the last transformed axis is ``s[-1]`` (if given) or ``2 * (m - 1)``, and all other axes ``s[i]``. The returned array must have a complex dtype with the same precision as the input ``x``.
+
+
+    **Raises**
+
+    - If ``s`` and ``axes`` have different lengths.
+    - If ``axes`` contains any invalid axis of ``a``.
+    """
+
+
+def hfft(a: array, /, *, n: Optional[int] = None, axis: int = -1, norm: str = 'backward') -> array:
+    """
+    Computes the one-dimensional discrete Fourier transform of a signal with Hermitian symmetry.
+
+    Parameters
+    ----------
+    a: array
+        input array. Should have a floating-point data type.
+    n: int
+        length of the transformed axis of the output. If given, the input will be either zero-padded or trimmed to this length before computing the Hermitian Fourier transform. Otherwise, it will default to ``2 * (m - 1)`` where ``m`` is the length of the input along the axis given by the ``axis`` keyword. Default: ``None``.
+    axis: int
+        axis used to compute the Fourier transform. If it is not specified, the last axis is used. Default: ``-1``.
+    norm: str
+        specify the normalization mode. Should be one of the following modes:
+
+        - ``'backward'``: No normalization.
+        - ``'ortho'``: Normalize by ``1/sqrt(n)``.
+        - ``'forward'``: Normalize by ``1/n``.
+
+        Default: ``'backward'``.
+
+    Returns
+    -------
+    out: array
+        a transformed array. The returned array must have a complex dtype with the same precision as the input ``x``.
```

Comment:
IIUC, the output array must have a real-valued floating-point data type and the input array should be a complex floating-point data type.